smoke-tests/apps/OpenTelemetryMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryMetricTestToAmw.java

```diff
@@ -0,0 +1,202 @@
+    List<DataPoint> dataPointList1 = md1.getMetrics();
+    assertThat(dataPointList1).hasSize(1);
+    DataPoint dp1 = dataPointList1.get(0);
+    assertThat(dp1.getValue()).isEqualTo(2.0);
+    assertThat(dp1.getName()).isEqualTo(name);
+
+    // validate custom dimension
+    Map<String, String> properties1 = md1.getProperties();
+    assertThat(properties1).containsEntry("tag1", "abc");
+    assertThat(properties1).containsEntry("tag2", "def");
+    assertThat(properties1).containsEntry("name", "apple");
+    assertThat(properties1).containsEntry("color", "green");
+    assertThat(properties1).containsEntry("_MS.SentToAMW", "True");
+
+    // validate tags
+    Map<String, String> tags2 = envelope2.getTags();
+    assertThat(tags2.get("ai.internal.sdkVersion")).isNotNull();
+    assertThat(tags2).containsEntry("ai.cloud.roleInstance", "testroleinstance");
+    assertThat(tags2).containsEntry("ai.cloud.role", "testrolename");
+    assertThat(tags2).containsEntry("ai.application.ver", "123");
+
+    // validate base data
+    MetricData md2 = (MetricData) ((Data<?>) envelope2.getData()).getBaseData();
+    List<DataPoint> dataPointList2 = md2.getMetrics();
+    assertThat(dataPointList2).hasSize(1);
+    DataPoint dp2 = dataPointList2.get(0);
+    assertThat(dp2.getValue()).isEqualTo(6.0);
+    assertThat(dp2.getName()).isEqualTo(name);
+
+    // validate custom dimension
+    Map<String, String> properties2 = md2.getProperties();
+    assertThat(properties2).containsEntry("tag1", "abc");
+    assertThat(properties2).containsEntry("tag2", "def");
+    assertThat(properties2).containsEntry("name", "apple");
+    assertThat(properties2).containsEntry("color", "red");
+    assertThat(properties2).containsEntry("_MS.SentToAMW", "True");
+
+    // validate tags
+    Map<String, String> tags3 = envelope3.getTags();
+    assertThat(tags3.get("ai.internal.sdkVersion")).isNotNull();
+    assertThat(tags3).containsEntry("ai.cloud.roleInstance", "testroleinstance");
+    assertThat(tags3).containsEntry("ai.cloud.role", "testrolename");
+    assertThat(tags3).containsEntry("ai.application.ver", "123");
+
+    // validate base data
+    MetricData md3 = (MetricData) ((Data<?>) envelope3.getData()).getBaseData();
+    List<DataPoint> dataPointList3 = md3.getMetrics();
+    assertThat(dataPointList3).hasSize(1);
+    DataPoint dp3 = dataPointList3.get(0);
+    assertThat(dp3.getValue()).isEqualTo(7.0);
+    assertThat(dp3.getName()).isEqualTo(name);
+
+    // validate custom dimension
+    Map<String, String> properties3 = md3.getProperties();
+    assertThat(properties3).containsEntry("tag1", "abc");
+    assertThat(properties3).containsEntry("tag2", "def");
+    assertThat(properties3).containsEntry("name", "lemon");
+    assertThat(properties3).containsEntry("color", "yellow");
+    assertThat(properties3).containsEntry("_MS.SentToAMW", "True");
+  }
+
+  private void validateOtlpMetricsReceived(String name) throws Exception {
```

Comment:
it doesn't look like `name` is used in this function

Reply:
Removed.

Reply:
can you confirm if it's correct to remove it, or if some validation on name should be added instead?

Reply:
I didn't find any reference to this variable.